whisperv6/message.go

```diff
@@ -76,7 +76,7 @@ type ReceivedMessage struct {
 	SymKeyHash   common.Hash // The Keccak256Hash of the key
 	EnvelopeHash common.Hash // Message envelope hash to act as a unique id
 
-	P2P bool
+	P2P bool // is set to true if this message was received from mail server.
```

Comment:
hmm, why is it called P2P? maybe we can call it something like `FromMailserver` or `IsFromMailserver`?

Reply:
Currently, in the whole Whisper implementation, `p2p` means a message received directly from another peer which in our setup means from a mail server pretty much always (in theory that might not be always true). So it's a superset of a statement `received from mail server`. To say a message was received from a mail server, we would need some more guarantees.

Reply:
yeah this is why i simply called it P2P, as this was already used across whisper codebase. tbh this whole flow is funny, but i don't think additional information that certain envelope was received from a mail server will make it less funny. 

i will rely on this flag to update index though, and if in some future we will be using p2p messages for something else it will lead to errors.

Reply:
okay, no need to rename then